From 861f1eff7722c735a0dbeda426714c06268ad0ef Mon Sep 17 00:00:00 2001
From: Arulselvan M <arulselvan.m@intel.com>
Date: Thu, 30 May 2013 16:14:58 +0530
Subject: [PORT FROM MAIN][JB] HDMI disconnect during multichannel playback fails

BZ: 58732

The direct output thread exits on disconnect. Hence
on subsequent playback in IHF, the stream config will
be new. So new track/ device needs to be opened.
Therefore return dead object in these scenarios.

Change-Id: I60c2534217a57ca70ba612537707152a887c185e
Orig-Change-Id: I3c8c2319ebddcea78c460a9d405d0d0006b1bfef
Signed-off-by: Jerome Anand <jerome.anand@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 services/audioflinger/Tracks.cpp |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/services/audioflinger/Tracks.cpp b/services/audioflinger/Tracks.cpp
index 5ac3129..bf99f66 100644
--- a/services/audioflinger/Tracks.cpp
+++ b/services/audioflinger/Tracks.cpp
@@ -634,6 +634,10 @@ status_t AudioFlinger::PlaybackThread::Track::start(AudioSystem::sync_event_t ev
             mState = state;
             triggerEvents(AudioSystem::SYNC_EVENT_PRESENTATION_COMPLETE);
         }
+    } else if (thread == 0) {
+      /*in case of direct output thread, on disconnect; the thread exits hence
+       we need to make sure the tracks are restored */
+        status = DEAD_OBJECT;
     } else {
         status = BAD_VALUE;
     }
-- 
1.7.4.1

